File: L1415/src/circ_models/python/run_systemic.py
import systemic as circ
from scipy.integrate import odeint
import numpy as np 
import matplotlib.pyplot as plt

t = np.linspace(0,4000,4001)
init = circ.init_state_values()
params = circ.init_parameter_values()
#params = circ.init_parameter_values(R1_s= 50)

pressure_inds = circ.monitor_indices('plv','p1s','p2s')
flow_inds = circ.monitor_indices('qart_s','q_mit')#,'q1_s')
Elv_ind = circ.monitor_indices('Elv')

#solve ODEs 
s = odeint(circ.rhs,init,t,(params,))

print s[-1,:]

#plot volumes
plt.plot(t,s[:,0],t,s[:,1],t,s[:,2])
plt.legend(['LV','sys1','sys2'])
plt.title('Volumes')

print(t.shape)

#compute pressures and flows
pressures = np.zeros((t.shape[0],len(pressure_inds)))
flows = np.zeros((t.shape[0],len(flow_inds)))
Elv = np.zeros(t.shape)
print flows.shape
for i in range(t.shape[0]):
        m = circ.monitor(s[i],t[i],params)
        pressures[i,:] = m[pressure_inds]
        flows[i,:] =m[flow_inds]
	Elv[i] = m[Elv_ind]
	

#plot time varying elastance
plt.figure()
plt.plot(t,Elv)
plt.title('Time varying LV elastance')

#plot pressures
plt.figure()
plt.plot(t,pressures[:,0],t,pressures[:,1],t,pressures[:,2])
plt.title('Pressures')
plt.legend(['LV','sys1','sys2'])

#plot flows
plt.figure()
plt.plot(t,flows[:,0],t,flows[:,1])#,t,flows[:,2])
plt.title('Flows')
plt.legend(['Aorta','Mitral'])

plt.show()

File: L1415/src/circ_models/matlab/run_systemic.m
[p,p_names] = systemic_init_parameters();
%p_names(25) 
%i = find(strcmp(p_names, 'SLmin'))
%p(i) = 2.5; %adjust SLmin for isometric twitch

m_names = systemic_monitored_names();
pressure_inds = [5,6,7];
flow_inds = [8,9,10];
Elv_ind = 3;

[init,state_names] = systemic_init_states();


[t,s] = ode15s(@systemic_rhs,[0,4000],init,[],p);
    
pressures = zeros(size(s,1),3);
flows = zeros(size(s,1),3);
Elv = zeros(size(s,1),1);
for i = 1:size(s,1)
   m = systemic_monitor(t(i),s(i,:),p);
   pressures(i,:) = m(pressure_inds);
   flows(i,:) = m(flow_inds);
   Elv(i) = m(Elv_ind);
end

figure(1)
plot(t,pressures)
title('Pressures')

figure(2)
plot(t,flows)
title('Flows')

figure(3) 
plot(t,Elv)
title('LV elastance')

